### PR TITLE
[linux] amlogic clean up

### PR DESCRIPTION
**Pull request by lrusak**
after doing some testing on my Wetek Hub I found that many of these methods are overly complex or not used at all.

I have cleaned it up a lot. I tested runtime on my Wetek Hub and playback of a few different files (including HEVC)

a lot of work could still be done to make it comply with our coding and formatting standards.

### CODE REVIEW

File: xbmc/utils/AMLUtils.cpp [hunk 1]
```diff
@@ -41,295 +41,200 @@
...
-  return has_h264_4k2k;
-}
 
-bool aml_support_vp9()
-{
-  static int has_vp9 = -1;
+  if (!SysfsUtils::HasRW("/sys/devices/system/cpu/cpu0/cpufreq/scaling_min_freq"))
+  {
+    CLog::Log(LOGERROR, "AML: no rw on /sys/devices/system/cpu/cpu0/cpufreq/scaling_min_freq");
+    return false;
```
Comment (stefansaraev) [inline xbmc/utils/AMLUtils.cpp]: nope. this changes behaviour. 

having cpufreq enabled is not a requirment for hardware decoding.

File: xbmc/utils/AMLUtils.cpp [hunk 2]
```diff
@@ -41,295 +41,200 @@
...
   {
-    CRegExp regexp;
-    regexp.RegComp("vp9:.*compressed");
-    std::string valstr;
-    if (SysfsUtils::GetString("/sys/class/amstream/vcodec_profile", valstr) != 0)
-      has_vp9 = 0;
-    else
-      has_vp9 = (regexp.RegFind(valstr) >= 0) ? 1 : 0;
+    CLog::Log(LOGERROR, "AML: no rw on /sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq");
+    return false;
```
Comment (stefansaraev) [inline xbmc/utils/AMLUtils.cpp]: same here

File: xbmc/utils/AMLUtils.cpp [hunk 3]
```diff
@@ -41,295 +41,200 @@
...
+  }
+
+  if (!SysfsUtils::HasRW("/sys/devices/system/cpu/cpu0/cpufreq/scaling_governor"))
+  {
+    CLog::Log(LOGERROR, "AML: no rw on /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor");
+    return false;
```
Comment (stefansaraev) [inline xbmc/utils/AMLUtils.cpp]: ^

File: xbmc/utils/AMLUtils.cpp [hunk 4]
```diff
@@ -41,295 +41,200 @@
...
+  if (aml_has_frac_rate_policy() && !SysfsUtils::HasRW("/sys/class/amhdmitx/amhdmitx0/frac_rate_policy"))
+  {
+    CLog::Log(LOGERROR, "AML: no rw on /sys/class/amhdmitx/amhdmitx0/frac_rate_policy");
+    return false;
+  }
+
+  if (!SysfsUtils::HasRW("/sys/module/di/parameters/bypass_prog"))
+  {
+    CLog::Log(LOGERROR, "AML: no rw on /sys/module/di/parameters/bypass_prog");
+    return false;
```
Comment (stefansaraev) [inline xbmc/utils/AMLUtils.cpp]: same here. also SysfsUtils::Set*() is never used on all these nodes, so perhaps they should be completely removed ?

**Comment by lrusak** [comment]
@stefansaraev is that better?

**Comment by stefansaraev** [comment]
lgtm.

**Comment by lrusak** [comment]
Closing because there is no reason to put lipstick on a pig :pig:

**Comment by peak3d** [comment]
1.) I hope that we can remove this crap as sonn as possible (gbm / V4L2)
2.) Platform dependend stuff should go on bigger changes into the platform folders
3.) Optimizations have been removed, we should keep kodi fast.
4.) Not finally clarified if functions / settings may be used outside (coreelec e.g.)

File: xbmc/settings/Settings.cpp [hunk 5]
```diff
@@ -571,10 +567,6 @@ bool CSettings::InitializeDefinitions()
 #elif defined(TARGET_LINUX)
   if (CFile::Exists(SETTINGS_XML_FOLDER "linux.xml") && !Initialize(SETTINGS_XML_FOLDER "linux.xml"))
     CLog::Log(LOGFATAL, "Unable to load linux-specific settings definitions");
-#if defined(HAS_LIBAMCODEC)
-  if (aml_present() && CFile::Exists(SETTINGS_XML_FOLDER "aml-linux.xml") && !Initialize(SETTINGS_XML_FOLDER "aml-linux.xml"))
-    CLog::Log(LOGFATAL, "Unable to load aml-linux-specific settings definitions");
```
[the remark this answers is not in the document]
Comment (MartijnKaijser) [reply xbmc/settings/Settings.cpp]: It's not in our tree anymore so we don't need it

**Pull request closed without merging**